crawl-ref/source/ng-setup.cc

```diff
@@ -235,6 +237,167 @@ static void _give_ammo(weapon_type weapon, int plus)
+    UNRAND_BOOTS_ASSASSIN,              // boots of the Assassin
+    UNRAND_LEAR,                        // Lear's hauberk
+    UNRAND_ZHOR,                        // skin of Zhor
+    UNRAND_SALAMANDER,                  // salamander hide armour
+    UNRAND_WAR,                         // gauntlets of War
+    UNRAND_RESISTANCE,                  // shield of Resistance
+    UNRAND_FOLLY,                       // robe of Folly
+    UNRAND_MAXWELL,                     // Maxwell's patent armour
+    UNRAND_DRAGONMASK,                  // mask of the Dragon
+    UNRAND_NIGHT,                       // robe of Night
+    UNRAND_DRAGON_KING,                 // armour of the Dragon King
+    UNRAND_ALCHEMIST,                   // hat of the Alchemist
+    UNRAND_FENCERS,                     // fencer's gloves
+    UNRAND_STARLIGHT,                   // cloak of Starlight
+    UNRAND_GONG,                        // shield of the Gong
+    UNRAND_RCLOUDS,                     // robe of Clouds
+    UNRAND_PONDERING,                   // hat of Pondering
+    UNRAND_DEMON_AXE,                   // obsidian axe
+    UNRAND_LIGHTNING_SCALES,            // lightning scales
+    UNRAND_BLACK_KNIGHT_HORSE,          // Black Knight's horse barding
+    UNRAND_AUTUMN_KATANA,               // autumn katana
+    UNRAND_DEVASTATOR,                  // shillelagh "Devastator"
+    UNRAND_WOE,                         // Axe of Woe
+    UNRAND_MOON_TROLL_LEATHER_ARMOUR,   // moon troll leather armour
+    UNRAND_DARK_MAUL,                   // dark maul
+    UNRAND_HIGH_COUNCIL,                // hat of the High Council
+    UNRAND_ARC_BLADE,                   // arc blade
+    UNRAND_SPELLBINDER,                 // demon whip "Spellbinder"
+    UNRAND_ORDER,                       // lajatang of Order
+    UNRAND_FIRESTARTER,                 // great mace "Firestarter"
+    UNRAND_ORANGE_CRYSTAL_PLATE_ARMOUR, // orange crystal plate armour
+    UNRAND_MAJIN,                       // Majin-Bo
+    UNRAND_GYRE,                        // pair of quick blades "Gyre" and "Gimble"
+    UNRAND_ETHERIC_CAGE,                // Maxwell's etheric cage
+    UNRAND_ETERNAL_TORMENT,             // crown of Eternal Torment
+    UNRAND_VINES,                       // robe of Vines
+    UNRAND_KRYIAS,                      // Kryia's mail coat
+    UNRAND_FROSTBITE,                   // frozen axe "Frostbite"
+    UNRAND_TALOS,                       // armour of Talos
+    UNRAND_WARLOCK_MIRROR,              // warlock's mirror
+};
+
+static skill_type _setup_archaeologist_crate(item_def& crate)
+{
+    item_def unrand;
+    unrand_type type;
+    do {
+        unrand = item_def();
+        type = _archa_unrands[random2(_archa_num_unrands)];
+        if (!make_item_unrandart(unrand, type))
+            continue;
+
+    } while ((!you.could_wield(unrand) && !can_wear_armour(unrand, false, true)) ||  !can_generate(unrand));
+
```

Comment:
For this `do-while` loop, you need to place the braces `{` and `}` on their own separate lines and format the `while` clause as follows for code readability (once again line too long):
`----while(!you.could_wield(unrand) && !can_wear_armour(unrand, false, true)`
`----------|| !can_generate(unrand));`
Note that the pair of brackets around the && section are not required.